src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -288,22 +295,22 @@ public BoundExpression Property(WellKnownMember member)
             return Property(null, member);
         }
 
-        public BoundExpression Property(BoundExpression receiverOpt, WellKnownMember member)
+        public BoundExpression Property(BoundExpression? receiverOpt, WellKnownMember member)
         {
-            var propertySym = (PropertySymbol)WellKnownMember(member);
+            var propertySym = (PropertySymbol)WellKnownMember(member)!;
```

Comment:
>(PropertySymbol)WellKnownMember(member)!; [](start = 30, length = 41)

how do we know? Feels like this should be an assertion instead

Reply:
`WellKnownMember` always returns non-null when not given its second argument.  I'll split into two methods and remove this annotation.

---
In reply to: [386585821](https://github.com/dotnet/roslyn/pull/42068#discussion_r386585821) [](ancestors = 386585821)